pyomo/contrib/pyros/tests/test_grcs.py

```diff
@@ -2126,6 +2129,90 @@ def test_terminate_with_time_limit(self):
         self.assertEqual(results.pyros_termination_condition, pyrosTerminationCondition.time_out,
                          msg="Returned termination condition is not return time_out.")
 
+    @unittest.skipUnless(SolverFactory('baron').license_is_valid(),
+                         "Global NLP solver is not available and licensed.")
+    def test_terminate_with_application_error(self):
+        """
+        Check that PyROS correctly raises ApplicationError
+        in event of abnormal IPOPT termination.
+        """
+        m = ConcreteModel()
+        m.p = Param(mutable=True, initialize=1.5)
+        m.x1 = Var(initialize=-1)
+        m.obj = Objective(expr=log(m.x1) * m.p)
+        m.con = Constraint(expr=m.x1 * m.p >= -2)
+
+        solver = SolverFactory("ipopt")
+        solver.options["halt_on_ampl_error"] = "yes"
+        baron = SolverFactory("baron")
+
+        box_set = BoxSet(bounds=[(1, 2)])
+        pyros_solver = SolverFactory("pyros")
+        with self.assertRaisesRegex(
+                ApplicationError,
+                r"Solver \(ipopt\) did not exit normally",
+                ):
+            pyros_solver.solve(
+                m,
+                [m.x1],
+                [],
+                [m.p],
+                box_set,
+                solver,
+                baron,
```

Comment:
With this many arguments, we should consider specifying them all as keyword arguments.

Reply:
Agreed